sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -346,3 +403,46 @@ rom_error_t boot_data_read(lifecycle_state_t lc_state, boot_data_t *boot_data) {
       return error;
   }
 }
+
+rom_error_t boot_data_write(const boot_data_t *boot_data) {
+  boot_data_t new_entry = *boot_data;
+  new_entry.identifier = kBootDataIdentifier;
+  boot_data_page_info_t active_page;
+  rom_error_t error = boot_data_active_page_find(&active_page);
+
+  if (error == kErrorOk) {
+    // Note: Not checking for wraparound since a successful write will
+    // invalidate the old entry.
+    new_entry.counter = active_page.last_valid_entry.counter + 1;
+    RETURN_IF_ERROR(boot_data_digest_compute(&new_entry, &new_entry.digest));
+    if (active_page.has_empty_entry == kHardenedBoolTrue) {
+      RETURN_IF_ERROR(boot_data_entry_write(
+          active_page.base_addr, active_page.first_empty_index, &new_entry));
+    } else {
+      uint32_t new_page_base = active_page.base_addr == kBootDataPage0Base
+                                   ? kBootDataPage1Base
+                                   : kBootDataPage0Base;
+      // TODO(#8777): Update error handling after switching to silicon_creator
+      // driver.
+      if (flash_page_erase(new_page_base, kInfoPartition) != 0) {
+        return kErrorBootDataFlash;
+      }
+      RETURN_IF_ERROR(boot_data_entry_write(new_page_base, 0, &new_entry));
+    }
+    RETURN_IF_ERROR(boot_data_entry_invalidate(active_page.base_addr,
```

Comment:
This probably deserves a comment. The way I interpret this is, on successful state transitions, there is only one valid entry between the two pages, and no fallback in case the active partition gets corrupted over time.

Reply:
Correct, added a comment.